### PR TITLE
Add documentation for @live functions

### PR DESCRIPTION
**Pull request by WalterBright**
Since it's in the compiler, it needs documentation.

### CODE REVIEW

**Comment by 12345swordy** [comment]
I think that @atilaneves needs admin access now, sense that andralex has step down from leadership.

File: spec/ob.dd [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+$(BLOCKQUOTE At any point in the program, for each memory object,
+there is exactly one live mutable pointer to it or all the live pointers to
+it are read-only.)
+
+$(P Function declarations annotated with the `@live` attribute are checked for compliance with
+the Ownership/Borrowing rules. The checks are run after other semantic processing is complete.
+The checking does not influence code generation.
+)
+
+$(P Whether a pointer is allocates memory using the GC or some other
```
Comment (baryluk) [inline spec/ob.dd:91]: Some grammatical error here with `is allocates`?

Comment (WalterBright) [reply spec/ob.dd]: fixed

File: spec/ob.dd [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+if non-`@live` functions call `@live` functions, arguments passed are
+expected to follow `@live` conventions.
+)
+
+$(P It will not detect attempts to dereference `null` pointers or possibly
+`null` pointers. This is unworkable because there is no current method
+of annotating a type as a non-`null` pointer.
+)
+
+$(H4 Tracked Pointers)
```
Comment (kubo39) [inline spec/ob.dd:112]: Is it H3, not H4?

**Pull request merged**